File: FormulaBolloApp/src/main/java/formula/bollo/entity/Penalty.java
package formula.bollo.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "penalty")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Penalty {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, length = 4)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "race_id")
    private Race race;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "driver_id")
    private Driver driver;

    @Column(name = "reason", nullable = false, length = 250)
    private String reason;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "penalty_severity_id")
    private PenaltySeverity penaltySeverity;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "season_id")
    private Season season;
}

File: FormulaBollo/src/shared/models/penalty.ts
import { Driver } from "./driver";
import { PenaltySeverity } from "./penaltySeverity";
import { Race } from "./race";
import { Season } from "./season";

export class Penalty {
  id: number;
  race: Race;
  driver: Driver;
  reason: string;
  penaltySeverity: PenaltySeverity;
  season: Season;

  constructor(id: number, race: Race, driver: Driver, reason: string, penaltySeverity: PenaltySeverity, season: Season) {
    this.id = id;
    this.race = race;
    this.driver = driver;
    this.reason = reason;
    this.penaltySeverity = penaltySeverity;
    this.season = season;
  }
}